lib/openhab/dsl/items/image_item.rb

```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+require 'base64'
+require 'net/http'
+require 'java'
+require 'mimemagic'
+require 'openhab/dsl/items/item_command'
+require 'openhab/dsl/items/item_delegate'
+
+module OpenHAB
+  module DSL
+    module Items
+      #
+      # Delegator to OpenHAB Player Item
+      #
+      class ImageItem
+        extend OpenHAB::DSL::Items::ItemCommand
+        extend OpenHAB::DSL::Items::ItemDelegate
+
+        def_item_delegator :@image_item
+
+        item_type Java::OrgOpenhabCoreLibraryItems::ImageItem
+
+        #
+        # Creates a new ImageItem
+        #
+        # @param [Java::OrgOpenhabCoreLibraryItems::ImageItem] image_item
+        #   The OpenHAB ImageItem to delegate to
+        #
+        def initialize(image_item)
+          logger.trace("Wrapping #{image_item}")
+          @image_item = image_item
+
+          item_missing_delegate { @image_item }
+
+          super()
+        end
+
+        #
+        #
+        # Update image with base64 encoded OpenHAB compatable image
+        #
+        # @param [String] base_64 base_64 encoding of an image
+        #
+        #
+        def update(base_64_encoded_image)
+          logger.trace { "Updating #{self} with Base64 image #{base_64_encoded_image}" }
+          @image_item.update base_64_encoded_image
+        end
+
+        #
+        # Update image from file
+        #
+        # @param [String] file location
+        # @param [String] mime_type of image
+        #
+        #
+        def update_from_file(file, mime_type: nil)
+          mime_type ||= detect_mime_type(file: file)
+          update_from_bytes(IO.binread(file), mime_type: mime_type)
```

Comment:
We should avoid reading the file twice (once inside inside detect_mime_type -> mime_from_file_contents)

```suggestion
          file_data = IO.binread(file)
          mime_type ||= (MimeMagic.by_path(file) || MimeMagic.by_magic(file_data))&.type
          update_from_bytes(file_data, mime_type: mime_type)
```

By this, we won't need detect_mime_type and mime_from_file_contents methods